test/Elastic.Apm.AspNetCore.Tests/AspNetCoreDiagnosticListenerTest.cs

```diff
@@ -53,20 +52,20 @@ public async Task TestErrorInAspNetCore(bool useOnlyDiagnosticSource)
 
 				capturedPayload.Should().NotBeNull();
 
-				capturedPayload.Transactions.Should().ContainSingle();
+				capturedPayload?.Transactions.Should().ContainSingle();
```

Comment:
The null conditionals aren't needed here IMO, as a null reference will fail the test on [line 53](https://github.com/elastic/apm-agent-dotnet/pull/982/files#diff-7c522236e53c6b6553577390af5e0b16cc269c08530571ec49216b7017ebecd9R53)

Reply:
I totally agree with you, but Rider is not smart enough and gives a warning if we don't care about the potential `null`:

![image](https://user-images.githubusercontent.com/1091853/97860642-ca884380-1d02-11eb-8bd5-7ac3c0787105.png)

I'd keep the `?.` that way we don't have warnings.

Reply:
unfortunate to code in a specific way to satisfy tooling 😃 I had a quick look at seeing if null-checking options can be configured for this, but doesn't look so. Maybe wrapping all in a `if(capturedPayload != null) { ... }` satisfies the warning _and_ only does the check once?

Reply:
Using `if` block would definitely work - but that's also coding in a specific way to satisfy tooling - Ultimately I would not null check here (as we agree it's not needed), but the warnings drive me crazy :)

I don't have strong feelings and I'd be happy to push a change to use `if`s - only thing is that I use `?.` for this in lots of other places.

Again, I'm fine with both `?.` and `if` blocks, but mixing the two would be maybe nice to avoid. Would you be ok with maybe changing this in a separate PR at all places in case we really want to do so?

Reply:
for sure! Totally not needed for this PR